File: 프로그래머스/lv2/42586. 기능개발/기능개발.java
import java.util.*;
class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        
        int curVer =0;
        int cnt=0;
        List<Integer> list  = new ArrayList<>();
        while(true){
            for(int i=0; i< progresses.length; i++){
                progresses[i] += speeds[i];
            }
            if(progresses[curVer] >=100){
                cnt=0;
                while(curVer < progresses.length &&progresses[curVer] >=100){
                    curVer ++;
                    cnt++;
                }
                list.add(cnt);
            }
            if(curVer == progresses.length){
                break;
            }
            
        }
        return list.stream().mapToInt(i->i).toArray();
    }
}

File: 프로그래머스/lv2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    while progresses:
        for idx in range(len(speeds)):
            progresses[idx] += speeds[idx]
        deploy = 0
        while progresses and progresses[0] >= 100:
            deploy +=1
            progresses.pop(0)
            speeds.pop(0)
        if deploy != 0:
            answer.append(deploy)
    return answer